tests/test_integration.py

```diff
@@ -235,155 +278,165 @@ def test_postboot(self):
             f.write('echo hello > /root/postboot_result')
             f.flush()
 
-            buildvm(self.vm.server_obj['hostname'], postboot=f.name)
-            self.assertEqual(self.vm.hypervisor.fqdn, self.hv.fqdn)
-            self._check_vm(self.hv, self.vm)
+            buildvm(self.vm_obj['hostname'], postboot=f.name)
+            self.check_vm_present()
 
             output = self.vm.run('cat /root/postboot_result')
             self.assertIn('hello', output)
 
     def test_delete(self):
-        buildvm(self.vm.server_obj['hostname'])
+        buildvm(self.vm_obj['hostname'])
+        self.check_vm_present()
 
         # Fails while VM is powered on
         with self.assertRaises(IGVMError):
-            vm_delete(self.vm.server_obj['hostname'])
+            vm_delete(self.vm_obj['hostname'])
 
         self.vm.shutdown()
-        vm_delete(self.vm.server_obj['hostname'], retire=True)
+        vm_delete(self.vm_obj['hostname'], retire=True)
 
-        self._check_absent(self.hv, self.vm)
+        self.check_vm_absent()
 
     def test_rollback(self):
-        self.vm.server_obj['puppet_environment'] = 'doesnotexist'
-        self.vm.server_obj.commit()
+        self.vm_obj['puppet_environment'] = 'doesnotexist'
+        self.vm_obj.commit()
 
         with self.assertRaises(IGVMError):
-            buildvm(self.vm.server_obj['hostname'])
+            buildvm(self.vm_obj['hostname'])
 
-        # Have we cleaned up?
-        self._check_absent(self.hv, self.vm)
+        self.check_vm_absent()
 
     def test_image_corruption(self):
```

Comment:
Maybe add a docstring, explaining that we expect igvm to redownload this image. I thought about this for a bit before realizing. Otherwise this test would require quite a bit of luck 🤔

Reply:
Done.